crypto/evp/evp_enc.c

```diff
@@ -365,8 +365,9 @@ static int evp_cipher_init_internal(EVP_CIPHER_CTX *ctx,
             ctx->num = 0;
             /* Don't reuse IV for CTR mode */
             if (iv != NULL) {
-                if ((n = EVP_CIPHER_CTX_get_iv_length(ctx)) <= 0)
-                    return 0;
+                n = EVP_CIPHER_CTX_get_iv_length(ctx);
+                if (!ossl_assert(n > 0 && n <= (int)sizeof(ctx->iv)))
```

Comment:
Previously we did not assert on n <= 0. But OK

Reply:
Earlier in the switch statement, there is a similar assert.  I figure it was appropriate here.

Reply:
Do you mean the assert in the `EVP_CIPH_CBC_MODE`? It was added a year ago (by you) in commit c12bf35026af94a73402eaf13f2428a9af30f1c0 to fix another coverity warning.

Reply:
It was an assert already, I just improved it :)

Reply:
Ok, so this particular location is not your fault ;-). Nevertheless, we're inconsistent.

Reply:
It is actually possible that `EVP_CIPHER_CTX_get_iv_length` may return -1. So an assert does not seem appropriate.

Reply:
That was exactly my point in [my review](https://github.com/openssl/openssl/pull/17898#pullrequestreview-922769641).